<br> <font size=7 color=#009999> <b> Tâche 5 - Lattice </b> <br><br>

<font size=6 color=#009999> <br> Imports <br> <br>

In [1]:
from pymatgen.core import Lattice, Structure, Molecule
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.ext.matproj import MPRester

<font size=6 color=#009999> <br> Téléchargement du fichier CIF du matériau mp-867168 ($Sr_2SbAu$) <br> <br>

In [2]:
with MPRester("V4zgRFcb4T97MTrH") as m:
    structure = m.get_structure_by_material_id("mp-867168")
structure = structure.get_primitive_structure(tolerance=0.1)
sga = SpacegroupAnalyzer(structure) 


<font size=6 color=#009999> <br> Description de la structure du $Sr_2SbAu$<br> <br>

<font size=5 color=#009999> <br> 1. Vecteurs du réseau de base <br> <br>

Le réseau de base d'un matériau est défini par $3$ vecteurs de base indépendants tel que tout point du réseau peut s'écrire : $\mathbf{R} = l \mathbf{a} + m \mathbf{b} + n \mathbf{c}$ avec $l,m,n \in \mathbb{Z}$. Ces vecteurs définissent une maille primitive ou conventionnelle de la structure du matériau. 

In [3]:
lattice = structure.lattice
vectorNorm = lattice.abc
angles = lattice.angles
volume = lattice.volume
vectors = lattice.matrix

for i in [0,1,2]:
    print("Vecteur du réseau deS base ", i+1, "\n")
    print("     Angle associé [degrés] : ", angles[i], "\n")
    print("     Norme: ", vectorNorm[i], "\n")
    print("     Norme selon la direction x : ", vectors[i][0], "\n")
    print("     Norme selon la direction y : ", vectors[i][1], "\n")
    print("     Norme selon la direction z : ", vectors[i][2], "\n")
        
print("Le volume total de la maille crée est de : ", volume, ". \n")

Vecteur du réseau deS base  1 

     Angle associé [degrés] :  59.99999999999999 

     Norme:  5.611844074442732 

     Norme selon la direction x :  0.0 

     Norme selon la direction y :  3.968173 

     Norme selon la direction z :  3.968173 

Vecteur du réseau deS base  2 

     Angle associé [degrés] :  59.99999999999999 

     Norme:  5.611844074442732 

     Norme selon la direction x :  3.968173 

     Norme selon la direction y :  0.0 

     Norme selon la direction z :  3.968173 

Vecteur du réseau deS base  3 

     Angle associé [degrés] :  59.99999999999999 

     Norme:  5.611844074442732 

     Norme selon la direction x :  3.968173 

     Norme selon la direction y :  3.968173 

     Norme selon la direction z :  0.0 

Le volume total de la maille crée est de :  124.968854511472 . 



Les informations affichées ci-dessus permet d'exprimer les vecteurs de base par : 
- $\mathbf{a} = 3.968173 y + 3.968173 z$
- $\mathbf{b} = 3.968173 x + 3.968173 z$
- $\mathbf{c} = 3.968173 x + 3.968173 y$

Les angles entre tous les vecteurs de base valent $\alpha=\beta=\gamma = 60^\circ$ et leur norme est de $|\mathbf{a}|=|\mathbf{b}|=|\mathbf{c}|=5.61$.

<font size=5 color=#009999> <br> 2. Vecteurs du réseau réciproque <br> <br>

Le réseau réciproque d'un matériau est défini par $3$ vecteurs de base indépendants tels que $e^{i\mathbf{K}\mathbf{R}}=1$ pour tout $\mathbf{R}$ du réseau direct et tel que tout point du réseau peut s'écrire : $\mathbf{K} = h \mathbf{a^*} + k \mathbf{b^*} + l \mathbf{c^*}$ avec $h,k,l \in \mathbb{Z}$.

In [4]:
reciprocalLattice = structure.lattice.reciprocal_lattice
reciprocalVectorNorm = reciprocalLattice.abc
reciprocalAngles = reciprocalLattice.angles
reciprocalVolume = reciprocalLattice.volume
reciprocalVectors = reciprocalLattice.matrix

for i in [0,1,2]:
    print("Vecteur du réseau réciproque ", i+1, "\n")
    print("     Angle associé [degrés] : ", reciprocalAngles[i], "\n")
    print("     Norme: ", reciprocalVectorNorm[i], "\n")
    print("     Norme selon la direction x : ", reciprocalVectors[i][0], "\n")
    print("     Norme selon la direction y : ", reciprocalVectors[i][1], "\n")
    print("     Norme selon la direction z : ", reciprocalVectors[i][2], "\n")
        
print("Le volume total de la maille crée est de : ", reciprocalVolume, ". \n")

Vecteur du réseau réciproque  1 

     Angle associé [degrés] :  109.47122063449069 

     Norme:  1.3712602985562001 

     Norme selon la direction x :  -0.7916975025004687 

     Norme selon la direction y :  0.7916975025004688 

     Norme selon la direction z :  0.7916975025004688 

Vecteur du réseau réciproque  2 

     Angle associé [degrés] :  109.47122063449069 

     Norme:  1.3712602985562001 

     Norme selon la direction x :  0.7916975025004688 

     Norme selon la direction y :  -0.7916975025004687 

     Norme selon la direction z :  0.7916975025004688 

Vecteur du réseau réciproque  3 

     Angle associé [degrés] :  109.47122063449069 

     Norme:  1.3712602985562006 

     Norme selon la direction x :  0.791697502500469 

     Norme selon la direction y :  0.791697502500469 

     Norme selon la direction z :  -0.791697502500469 

Le volume total de la maille crée est de :  1.984896272051752 . 



Les informations affichées ci-dessus permet d'exprimer les vecteurs de base par : 
- $\mathbf{a^*} = -0.791697 x + 0.791697 y + 0.791697 z$
- $\mathbf{b^*} = 0.791697 x - 0.791697 y + 0.791697 z$
- $\mathbf{c^*} = 0.791697 x + 0.791697 y - 0.791697 z$

Les angles entre tous les vecteurs de base valent $\alpha=\beta=\gamma = 109.471220^\circ$ et leur norme est de $|\mathbf{a^*}|=|\mathbf{b^*}|=|\mathbf{c^*}|=1.371260$

<font size=5 color=#009999> <br> 3. Type de maille <br> <br>

In [5]:
sga.get_lattice_type()

'cubic'

Ceci signifie que le type de maille est cubique.

<font size=5 color=#009999> <br> 4. Système cristallin <br> <br>

In [6]:
sga.get_crystal_system()

'cubic'

Ceci signifie que le type de système cristallin est cubique.

<font size=5 color=#009999> <br> 5. Groupe ponctuel <br> <br>

In [7]:
sga.get_point_group_symbol()

'm-3m'

Ceci signifie que son groupe ponctuel est m-3m.